### DISCUSSION TITLE
Practical Deep Learning Study Group

### DISCUSSION THREAD

**Discussion by Jenna59** [Virtual-Coffee/virtualcoffee.io#1223] [Technical Resources]
[Practical Deep Learning for Coders 2022 part 1](https://course.fast.ai/), recorded at the University of Queensland, covers topics such as how to:

- Build and train deep learning models for computer vision, natural language processing, tabular analysis, and collaborative filtering problems
- Create random forests and regression models
- Deploy models
- Use PyTorch, the world’s fastest growing deep learning software, plus popular libraries like fastai and Hugging Face

## What you will learn
After finishing this course you will know:
- How to train models that achieve state-of-the-art results in:
- Computer vision, including image classification (e.g., classifying pet photos by breed)
- Natural language processing (NLP), including document classification (e.g., movie review sentiment analysis) and phrase similarity
- Tabular data with categorical data, continuous data, and mixed data
- Collaborative filtering (e.g., movie recommendation)
- How to turn your models into web applications, and deploy them
- Why and how deep learning models work, and how to use that knowledge to improve the accuracy, speed, and reliability of your models
- The latest deep learning techniques that really matter in practice
- How to implement stochastic gradient descent and a complete training loop from scratch

### 1: Getting started - Tues. Sept 24th

Suggested Reading: [Ch #1](https://github.com/fastai/fastbook/blob/master/01_intro.ipynb)

### 2: Deployment - Tues. Oct 8th

Suggested Reading: [Ch #2](https://github.com/fastai/fastbook/blob/master/02_production.ipynb)

### 3: Neural net foundations - Tues. Oct 22nd

Suggested Reading: [Ch #4](https://github.com/fastai/fastbook/blob/master/04_mnist_basics.ipynb)

### 4: Natural Language Processing (NLP) - Tues. Nov 5th

Suggested Reading: [Ch#10](https://github.com/fastai/fastbook/blob/master/10_nlp.ipynb)

### 5: From-scratch model - Tues. Nov 19th

Suggested Reading: [Ch #4](https://github.com/fastai/fastbook/blob/master/04_mnist_basics.ipynb) & [Ch #9](https://github.com/fastai/fastbook/blob/master/09_tabular.ipynb)

### 6: Random forests - Tues. Dec 3rd

Suggested Reading: [Ch #9](https://github.com/fastai/fastbook/blob/master/09_tabular.ipynb)

### 7: Collaborative filtering - Tues. Dec 17th

Suggested Reading: [Ch #8](https://github.com/fastai/fastbook/blob/master/08_collab.ipynb)

### 8: Convolutions (CNNs) - Tues. Jan 7th

Suggested Reading: [Ch #13](https://github.com/fastai/fastbook/blob/master/13_convolutions.ipynb)

### 9: Bonus: Data Ethics - Tues. Jan 21st

Suggested Reading: [Ch #3](https://github.com/fastai/fastbook/blob/master/03_ethics.ipynb)

**Comment by Jenna59** [comment]
## Week 1 Discussion Questions

1. Do you need these for deep learning?
* Lots of math T / F
* Lots of data T / F
* Lots of expensive computers T / F
* A PhD T / F

2. Draw a picture that summarizes Samuel's view of a machine learning model.

3. What is a validation set? What is a test set? Why do we need them?

4. What is overfitting? Provide an example.

5. What kinds of features do the early layers of a CNN find? How about the later layers?

**Reply by virtualandy** [reply]
Good questions! I will take a crack when I can re-review materials.

Another question for folks:
* Were you able to get a Notebook running with some of the examples? Either locally or on [Google Colab]() or [Kaggle](https://www.kaggle.com/code/jhoward/saving-a-basic-fastai-model)?

If not, post here and someone can try to help you get setup.
